docs/user/developing_jobs.md

```diff
@@ -4,224 +4,155 @@ A goal of this plugin is to make it relatively quick and straightforward to deve
+from your_ssot_app.models import DiffSyncDevice, DiffSyncPrefix, DiffSyncIPAddress
 
-```python
-class DataSource(DataSource, Job):
-    ...  # Excluded most of the class definition for example brevity
 
-    def execute_sync(self):
-        self.log_info(obj=None, message="The actual sync happens in post_run to escape the atomic transaction.")
+class NautobotAdapter(NautobotAdapter):
+    top_level = ("device", "prefix")
 
-    def post_run(self):
-        super().execute_sync()
+    device = DiffSyncDevice
+    prefix = DiffSyncPrefix
+    ip_address = DiffSyncIPAddress  # Not in the `top_level` tuple, since it's a child of the prefix model
 ```
 
-Using this example, the CRUD (create, update and delete) operations of your job will not happen as part of the atomic database transaction, because `post_run` is run outside of that. This brings with it the following caveats:
+The `load` function is already implemented on this adapter and will automatically and recursively traverse any children relationships for you, provided the models are defined correctly.
 
-- The job result page in Nautobot will show the status of "Completed" _before_ your actual sync runs
-  - You will need to manually update the page to get further job log results
-  - The job result might still go into an erroneous state, updating the job result status
-- If you implement any further logic in `post_run` keep in mind that it doesn't bubble exceptions up to the job result page in Nautobot
-- If any exceptions are encountered during the CRUD operations (i.e. the diffsync models' `create`, `update` and `delete` methods) they will _not_ trigger a rollback of the objects created during this job
+### Step 2.2 - Creating the Remote Adapter
 
-Due to these caveats it is recommended that you check carefully whether this optimization actually benefits your use case or not before applying it in production code.
+Regardless of which direction you are synchronizing data in, you need to write the `load` method for the remote adapter yourself. You can find many examples of how this can be done in the `nautobot_ssot.integrations` module, which contains pre-existing integrations with remote systems.
 
 !!! note
-    In Nautobot 2.0, jobs will no longer be atomic by default so this section will not apply anymore.
-
-#### Using Bulk ORM Operations
-
-Bulk ORM operations are available within Django to perform multiple inserts/updates in a single DB query. Take a look at the documentation for [bulk_create](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create) for example. These bulk operations should only be used very carefully in Nautobot. The following caveats apply when using them:
+As the features in `nautobot_ssot.contrib` are still very new, most existing integrations do not use them.
```

Comment:
I feel like we should at least be updating the Example Job to utilize the Mixin since it's supposed to represent our recommendation around how to create an SSoT App.

Reply:
I have done this for the example integration, added a link here.

Reply:
Not a link but a reference rather